bin/table_georges_comparison

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python2.7
+
+"""
+Create a LaTeX summary comparing annotated Krun results to those reported by
+the method described in Georges et. al. (2007).
+"""
+
+import argparse
+import numpy
+import os
+import os.path
+import sys
+
+sys.path.insert(0, os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))))
+from warmup.krun_results import parse_krun_file_with_changepoints
+from warmup.latex import preamble, end_document, end_table
+from warmup.latex import machine_name_to_macro, STYLE_SYMBOLS
+
+
+_TITLE = 'Comparison with Georges et. al. (2007)'
+_START_TABLE = lambda format_, headings: """
+{
+\\begin{tabular}{%s}
+\\toprule
+%s \\\\
+\\midrule
+""" % (format_, headings)
+
+
+def _start_table(format_, headings):
+    """Unlike the start_table() function in warmup.latex, this one does not
+    contain any code to configure sparklines.
+    """
+
+    return _START_TABLE(format_, headings)
+
+
+def _cov(data):
+    """Computes the coefficient of variation of data."""
+
+    return numpy.std(data) / numpy.mean(data)
+
+
+def compute_comparison(in_files, threshold):
+    """For each classification type, find the number of process executions that
+    Georges et. al. (2007) would have classified as having a steady state.
+    """
+
+    classifier, data_dictionaries = parse_krun_file_with_changepoints(in_files)
+    steady = classifier['steady']  # Min iterations expected in steady state.
+    counts = dict()
+    for machine in data_dictionaries:
+        counts[machine] = {'warmup': 0, 'slowdown': 0, 'flat': 0,
+                            'no steady state': 0, 'total warmup': 0,
+                            'total flat': 0, 'total slowdown': 0,
+                            'total no steady state': 0}
+        keys = sorted(data_dictionaries[machine]['wallclock_times'].keys())
+        for key in sorted(keys):
+            wallclock_times = data_dictionaries[machine]['wallclock_times'][key]
+            if len(wallclock_times) == 0:
+                print('WARNING: Skipping: %s from %s (no executions)' %
+                       (key, machine))
+            elif len(wallclock_times[0]) == 0:
+                print('WARNING: Skipping: %s from %s (benchmark crashed)' %
+                      (key, machine))
+            else:
+                bench, vm, variant = key.split(':')
+                n_pexecs = len(data_dictionaries[machine]['wallclock_times'][key])
+                for pexec in xrange(n_pexecs):
+                    stop = len(data_dictionaries[machine]['wallclock_times'][key][pexec]) - steady
+                    classification = data_dictionaries[machine]['classifications'][key][pexec]
+                    counts[machine]['total ' + classification] += 1
+                    for iteration in xrange(len(data_dictionaries[machine]['wallclock_times'][key][pexec])):
```

Comment:
We need to think a bit about this, as we're imposing our idea of when the steady state might be onto Georges. For example, if there are 2000 iterations, we say a steady state must have happened by 1500. It *could* be argued that Georges might still find a steady state somewhere between 1500 and 2000. My initial inclination is to give them the benefit of the doubt. Any thoughts?